packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -980,13 +981,15 @@ export class TokenBridge {
 
     const tokensRes: Record<FungibleToken, FlowRateInfoItem> = {};
 
-    // @note it's i + 2 because the first 2 promises are not token buckets
+    // @note we use the shifter to navigate the promise all response correctly
     for (let i = 0, l = req.tokens.length; i < l; i++) {
+      const shifter = (i + 1) * 2;
```

Comment:
yeah when i was building this feature i noticed that we weren't returning the `largeTransferThresholds` so I've added those to the promise all for each token. So the first 2 promises in the promise all array are for the `withdrawalQueueActivated` and `withdrawalDelay` then each token has a set of 2 promises after that for `flowRateBuckets` and `largeTransferThresholds`.

This shifter of (i + 1) * 2 gets us to the right place in the promise all for the 2 promises we want to retrieve when we're looping through by token.

Actually i might change this, its pretty messy. i will read the first 2 then splice them off the array so we just loop through with a shifter of 2i rather than this more complicated one so it's easier to read.

Reply:
done

Reply:
Thanks, this makes sense now